xml/System.Security.Cryptography/CryptoStream.xml

```diff
@@ -59,13 +59,13 @@
 >  This type implements the <xref:System.IDisposable> interface. When you have finished using the type, you should dispose of it either directly or indirectly by calling its <xref:System.Security.Cryptography.CryptoStream.Clear%2A> method, which in turn calls its <xref:System.IDisposable> implementation. To dispose of the type directly, call its <xref:System.Security.Cryptography.CryptoStream.Clear%2A> method in a `try`/`catch` block. To dispose of it indirectly, use a language construct such as `using` (in C#) or `Using` (in Visual Basic). For more information, see the "Using an Object that Implements IDisposable" section in the <xref:System.IDisposable> interface topic.  
   
 > [!IMPORTANT]
->  From .NET 6 onwards, a breaking change is implemented on reading <xref:System.Security.Cryptography.CryptoStream>. When `Stream.Read` or `Stream.ReadAsync` is called on one of the affected stream types with a buffer of length `N`, the operation completes when:
+ When `Stream.Read` or `Stream.ReadAsync` is called with a buffer of length `N`, the operation completes when either:
+ > -   At least one byte has been read from the stream, or
+ > -   The underlying stream that it wraps returns 0 from a call to read, indicating no more data is available.
+ Also, when `Stream.Read` or `Stream.ReadAsync` is called with a buffer of length 0, the operation succeeds once a call with a non-zero buffer would succeed.
 
-> - At least one byte has been read from the stream, or
-> - The underlying stream they wrap returns 0 from a call to its read, indicating no more data is available.
+Previous to .NET 6.0, `Stream.Read` and `Stream.ReadAsync` would not return until `N` bytes had been read from the stream, or the underlying stream returned 0 from a call to read. Code that assumed they would not return until all `N` bytes were read could fail to read all the content. See (this document)[https://docs.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams] for more information.
```

Comment:
```suggestion
Prior to .NET 6, `Stream.Read` and `Stream.ReadAsync` did not return until `N` bytes had been read from the stream or the underlying stream returned 0 from a call to `Read`. If your code assumed they wouldn't return until all `N` bytes were read, it could fail to read all the content. For more information, see (Partial and zero-byte reads in streams)[/dotnet/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams].
```